pkg/apis/monitoring/v1alpha1/alertmanager_config_types.go

```diff
@@ -770,6 +772,39 @@ type SNSConfig struct {
 	HTTPConfig *HTTPConfig `json:"httpConfig,omitempty"`
 }
 
+// TelegramConfig configures notifications via Telegram.
+// See https://prometheus.io/docs/alerting/latest/configuration/#telegram_config
+type TelegramConfig struct {
+	// Whether to notify about resolved alerts.
+	// +optional
+	SendResolved *bool `json:"sendResolved,omitempty"`
+	// The Telegram API URL i.e. https://api.telegram.org.
+	// If not specified, default API URL will be used.
+	// +optional
+	APIURL string `json:"apiURL,omitempty"`
+	// Telegram bot token
+	// The secret needs to be in the same namespace as the AlertmanagerConfig
+	// object and accessible by the Prometheus Operator.
+	// +kubebuilder:validation:Required
+	BotToken *v1.SecretKeySelector `json:"botToken,omitempty"`
```

Comment:
I don't think that `Required` will do anything, especially given that BotToken is a pointer which can be nil.

Looking more at the current code, I feel that we shouldn't use `v1.SecretKeySelector` for "mandatory" secret key selectors... This is the current definition from `k8s.io/api/core/v1/types.go`

```golang
// SecretKeySelector selects a key of a Secret.
// +structType=atomic
type SecretKeySelector struct {
        // The name of the secret in the pod's namespace to select from.
        LocalObjectReference `json:",inline" protobuf:"bytes,1,opt,name=localObjectReference"`
        // The key of the secret to select from.  Must be a valid secret key.
        Key string `json:"key" protobuf:"bytes,2,opt,name=key"`
        // Specify whether the Secret or its key must be defined
        // +optional
        Optional *bool `json:"optional,omitempty" protobuf:"varint,3,opt,name=optional"`
}
```

In this case, the Optional field means nothing because we always want the secret's key to exist. And if we had our own type, we could enforce that Name and Key fields aren't empty.

```golang
type SecretKeySelector struct {
        // The name of the secret in the object's namespace to select from.
        // +kubebuilder:validation:MinLength=1
        Name string 
        // The key of the secret to select from.  Must be a valid secret key.
        // +kubebuilder:validation:MinLength=1
        Key string `json:"key"`
}
```

Reply:
removed the Required attribute in the botToken attribute